src/TextObjects/Tag.ts

```diff
@@ -0,0 +1,138 @@
+import { TextDocument, Position, Range } from 'vscode'
+import {TextObject} from './TextObject';
+
+type Tag = { name: string; type: 'close' | 'open'; start: number; end: number };
+type MatchedTag = {
+    tag: string;
+    openingTagStart: number;
+    openingTagEnd: number;
+    closingTagStart: number;
+    closingTagEnd: number;
+};
+
+export class TextObjectTag extends TextObject {
+
+    TAG_REGEX = /\<(\/)?([^\>\<\s\/]+)(?:[^\>\<]*?)(\/)?\>/g
```

Comment:
I think `<` and `>` don't need to be escaped?
```ts
/<(\/)?([^<>\s\/]+)(?:[^<>]*?)(\/)?>/g
```

Reply:
Will this work if there are line breaks?

Reply:
Great points here! Assume you're talking about a line break on the opening tag itself? Will do some testing and see what comes up for these.

Reply:
Yes, line breaks inside the tag itself. :)

Reply:
Removed the escaping and added a handful of tests for various "line break inside tag" scenarios.